### PR TITLE
Retry connecting to device in CI after lost connection

### PR DESCRIPTION
**Pull request by vashworth**
Sometimes `ios-deploy` loses connection to the device after installing, starting debugserver, and launching. This is shown with an error message like:
```
Process 579 exited with status = -1 (0xffffffff) lost connection
```
This happens frequently in our CI system: https://github.com/flutter/flutter/issues/120808

Usually in CI, on retry it'll work and pass - so this is an attempt to retry without failing the test first. It's not guaranteed to fix since we're unable to recreate this error locally.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by vashworth** [comment]
Google testing seems to be failing on something unrelated. I tried clicking "Re-run" but hasn't done anything yet. I'll give it a little time and if not, I'll update the branch to trigger it to rerun.

File: packages/flutter_tools/lib/src/ios/devices.dart [hunk 1]
```diff
@@ -537,34 +537,12 @@ class IOSDevice extends Device {
...
-            deviceLogReader.debuggerStream = iosDeployDebugger;
-          }
-        }
-        // Don't port foward if debugging with a wireless device.
-        vmServiceDiscovery = ProtocolDiscovery.vmService(
-          deviceLogReader,
-          portForwarder: isWirelesslyConnected ? null : portForwarder,
-          hostPort: debuggingOptions.hostVmServicePort,
-          devicePort: debuggingOptions.deviceVmServicePort,
+        vmServiceDiscovery = _setupDebuggerAndVmServiceDiscovery(
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/ios/devices.dart:540]: from a cursory look, this diff doesn't change anything, just factors out the left side logic to a helper method, is that right?

Comment (vashworth) [reply packages/flutter_tools/lib/src/ios/devices.dart:540]: Pretty much, only difference is I added a `skipInstall` option

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by christopherfujino** [comment]
BTW, your google testing failure looks like a bug with FRoB. I chatted with the release team who is on it, I think they are going to put out a PSA soon.

**Comment by vashworth** [comment]
> BTW, your google testing failure looks like a bug with FRoB. I chatted with the release team who is on it, I think they are going to put out a PSA soon.

Thanks for the heads up!

**Pull request merged**